### DISCUSSION TITLE
Not able to connect broker from web with wss

### DISCUSSION THREAD

**Discussion by hgraghu** [dotnet/MQTTnet#1482] [Q&A]
Hi, 
I have deployed the MQTT broker on AWS EC2 instance, and I have deployed my web app (Asp.net core with angular) on the same instance. I'm using paho as the client where I'm able to connect broker only with HTTP and WS .I request how to setup the certificate to work on HTTPS and WSS.

**Comment by ShawnStoddard** [comment]
It will depend on what web server you are using.  Can you tell us more about the application, specifically what server and which ports are being used?

**Comment by hgraghu** [comment]
Hi,
Thanks for the response.I have hosted my web app on the IIS webserver on port 80,
For the Broker,I have placed the published bundle and ran the .exe, in the broker I have set port 443 as ("WithEncryptedEndpointPort")
Host.CreateDefaultBuilder(args)
            .ConfigureWebHostDefaults(webBuilder =>
            {
                webBuilder.UseKestrel(
                    o =>
                    {
                        o.ListenAnyIP(443, l => l.UseMqtt()); // MQTT pipeline
                        o.ListenAnyIP(5000); // Default HTTP pipeline
                    });

                webBuilder.UseStartup<Startup>();
            });

in the startup.cs
 public void ConfigureServices(IServiceCollection services)
    {
        services.AddHostedMqttServer(
        mqttServer => mqttServer.WithoutDefaultEndpoint()
        .WithEncryptedEndpoint()
        .WithEncryptedEndpointPort(443)
        .WithEncryptionCertificate(GetCertificate().Export(X509ContentType.Pfx))
        .WithEncryptionSslProtocol(SslProtocols.Tls12)
        .WithConnectionValidator(new MqttServerConnectionValidatorDelegate(ValidateConnection))
    )
I'm not sure how to pass the selfsigned certificate which I have set to the broker across webapp.
